lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,10 +71,16 @@ def serializable_hash_for_single_resource(options)
         end
 
         def resource_identifier_type_for(serializer)
-          if ActiveModel::Serializer.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
+          unless serializer.class._type.nil?
+            serializer.class._type
           else
```

Comment:
Avoid `unless` with an `else` clause. Rewrite as an `if` with the positive clause first (https://github.com/bbatsov/ruby-style-guide#no-else-with-unless)